pkg/core/consensus/reduction/firststep/step.go

```diff
@@ -94,14 +94,21 @@ func (p *Phase) Run(ctx context.Context, queue *consensus.Queue, evChan chan mes
 	}
 
 	p.handler = reduction.NewHandler(p.Keys, r.P, r.Seed)
-	// first we send our own Selection
-	if p.handler.AmMember(r.Round, step) {
-		m, _ := p.SendReduction(r.Round, step, &p.selectionResult.Candidate)
 
-		// Queue my own vote to be registered locally
-		evChan <- m
+	// send our own Selection
+	var wg sync.WaitGroup
+
+	a := reduction.NewAsyncSend(p.Reduction, r.Round, step, &p.selectionResult.Candidate)
+
+	if p.handler.AmMember(r.Round, step) {
+		_ = a.Go(ctx, &wg, reductionChan, reduction.Republish)
```

Comment:
Shall the returned cancel function be deferred?

Reply:
In 1th_reduction step, It is not deferred on purpose.